cognite/client/data_classes/relationships.py

```diff
@@ -1,139 +1,126 @@
-        relationship_type (str): Type of the relationship in order to distinguish between different relationships. In general relationship types should reflect references as they are expressed in natural sentences.  E.g. a flow through a pipe can be naturally represented by a `flowsTo`-relationship. On the other hand an alternative asset hierarchy can be represented with the `isParentOf`-relationship. The `implements`-relationship is intended to reflect references between a functional asset hierarchy and its implementation.
+        external_id (str): External id of the relationship, must be unique within the project.
+        source_external_id (str): External id of the CDF resource that constitutes the relationship source.
+        source_type (str): The CDF resource type of the relationship source. Must be one of the specified values.
+        target_external_id (str): External id of the CDF resource that constitutes the relationship target.
+        target_type (str): The CDF resource type of the relationship target. Must be one of the specified values.
+        start_time (int): Time, in milliseconds since Jan. 1, 1970, when the relationship became active. If there is no startTime, relationship is active from the beginning of time until endTime.
+        end_time (int): Time, in milliseconds since Jan. 1, 1970, when the relationship became inactive. If there is no endTime, relationship is active from startTime until the present or any point in the future. If endTime and startTime are set, then endTime must be strictly greater than startTime.
+        confidence (float): Confidence value of the existence of this relationship. Generated relationships should provide a realistic score on the likelihood of the existence of the relationship. Relationships without a confidence value can be interpreted at the discretion of each project.
+        data_set_id (Dict[str, Any]): The id or externalId of the dataset this relationship belongs to.
+        labels (List[Label]): A list of the labels associated with this resource item.
         created_time (int): Time, in milliseconds since Jan. 1, 1970, when this relationship was created in CDF.
         last_updated_time (int): Time, in milliseconds since Jan. 1, 1970, when this relationship was last updated in CDF.
         cognite_client (CogniteClient): The client to associate with this object.
     """
 
     def __init__(
         self,
-        source: Dict[str, Any] = None,
-        target: Dict[str, Any] = None,
+        external_id: str = None,
+        source_external_id: str = None,
+        source_type: str = None,
+        target_external_id: str = None,
+        target_type: str = None,
         start_time: int = None,
         end_time: int = None,
         confidence: float = None,
-        data_set: str = None,
-        external_id: str = None,
-        relationship_type: str = None,
+        data_set_id: Dict[str, Any] = None,
```

Comment:
This doesn't seem to be correct. The API expects this to be just a number, not an object. The documentation and I tried myself as well. See https://pr-1019.specs.preview.cogniteapp.com/v1.json.html#tag/Relationships/paths/~1api~1v1~1projects~1{project}~1relationships~1list/post. 

You should add a test where you use data_set_id when creating a relationship.

Reply:
This is also consistent with our other APIs, though I think it's weird that we don't support setting it directly from an externalId when creating, while being able to filter just by the externalId.

Reply:
Good catch! I didn't add a test for datasetId because there's no way to delete them and so every time a new datasetId would get created but yeah there should be one.
And we can filter both by external and internal id, if I understood your statement correctly.